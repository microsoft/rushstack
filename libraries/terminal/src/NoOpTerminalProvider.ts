// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import type { ITerminalProvider, TerminalProviderSeverity } from './ITerminalProvider';

/**
 * Terminal provider that stores written data in buffers separated by severity.
 * This terminal provider is designed to be used when code that prints to a terminal
 * is being unit tested.
 *
 * @beta
 */
export class NoOpTerminalProvider implements ITerminalProvider {
  /**
   * {@inheritDoc ITerminalProvider.write}
   */
  public write(data: string, severity: TerminalProviderSeverity): void {
    // no-op
  }

  /**
   * {@inheritDoc ITerminalProvider.eolCharacter}
   */
  public get eolCharacter(): string {
    return '\n';
  }

  /**
   * {@inheritDoc ITerminalProvider.supportsColor}
   */
  public get supportsColor(): boolean {
    return false;
  }
}
