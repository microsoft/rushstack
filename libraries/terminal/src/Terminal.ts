// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { type ITerminalProvider, TerminalProviderSeverity } from './ITerminalProvider';
import { Colorize } from './Colorize';
import type { ITerminal, ITerminalWriteOptions, TerminalWriteParameters } from './ITerminal';
import { AnsiEscape } from './AnsiEscape';

/**
 * This class facilitates writing to a console.
 *
 * @beta
 */
export class Terminal implements ITerminal {
  private readonly _providers: Set<ITerminalProvider>;

  public constructor(provider: ITerminalProvider) {
    this._providers = new Set<ITerminalProvider>([provider]);
  }

  /**
   * {@inheritdoc ITerminal.registerProvider}
   */
  public registerProvider(provider: ITerminalProvider): void {
    this._providers.add(provider);
  }

  /**
   * {@inheritdoc ITerminal.unregisterProvider}
   */
  public unregisterProvider(provider: ITerminalProvider): void {
    this._providers.delete(provider);
  }

  /**
   * {@inheritdoc ITerminal.write}
   */
  public write(...messageParts: TerminalWriteParameters): void {
    const { parts } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(parts, TerminalProviderSeverity.log, false);
  }

  /**
   * {@inheritdoc ITerminal.writeLine}
   */
  public writeLine(...messageParts: TerminalWriteParameters): void {
    const { parts } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(parts, TerminalProviderSeverity.log, true);
  }

  /**
   * {@inheritdoc ITerminal.writeWarning}
   */
  public writeWarning(...messageParts: TerminalWriteParameters): void {
    const {
      parts,
      options: { doNotOverrideSgrCodes }
    } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(
      doNotOverrideSgrCodes
        ? parts
        : parts.map((part): string => Colorize.yellow(AnsiEscape.removeCodes(part))),
      TerminalProviderSeverity.warning,
      false
    );
  }

  /**
   * {@inheritdoc ITerminal.writeWarningLine}
   */
  public writeWarningLine(...messageParts: TerminalWriteParameters): void {
    const {
      parts,
      options: { doNotOverrideSgrCodes }
    } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(
      doNotOverrideSgrCodes
        ? parts
        : parts.map((part): string => Colorize.yellow(AnsiEscape.removeCodes(part))),
      TerminalProviderSeverity.warning,
      true
    );
  }

  /**
   * {@inheritdoc ITerminal.writeError}
   */
  public writeError(...messageParts: TerminalWriteParameters): void {
    const {
      parts,
      options: { doNotOverrideSgrCodes }
    } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(
      doNotOverrideSgrCodes ? parts : parts.map((part): string => Colorize.red(AnsiEscape.removeCodes(part))),
      TerminalProviderSeverity.error,
      false
    );
  }

  /**
   * {@inheritdoc ITerminal.writeErrorLine}
   */
  public writeErrorLine(...messageParts: TerminalWriteParameters): void {
    const {
      parts,
      options: { doNotOverrideSgrCodes }
    } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(
      doNotOverrideSgrCodes ? parts : parts.map((part): string => Colorize.red(AnsiEscape.removeCodes(part))),
      TerminalProviderSeverity.error,
      true
    );
  }

  /**
   * {@inheritdoc ITerminal.writeVerbose}
   */
  public writeVerbose(...messageParts: TerminalWriteParameters): void {
    const { parts } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(parts, TerminalProviderSeverity.verbose, false);
  }

  /**
   * {@inheritdoc ITerminal.writeVerboseLine}
   */
  public writeVerboseLine(...messageParts: TerminalWriteParameters): void {
    const { parts } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(parts, TerminalProviderSeverity.verbose, true);
  }

  /**
   * {@inheritdoc ITerminal.writeDebug}
   */
  public writeDebug(...messageParts: TerminalWriteParameters): void {
    const { parts } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(parts, TerminalProviderSeverity.debug, false);
  }

  /**
   * {@inheritdoc ITerminal.writeDebugLine}
   */
  public writeDebugLine(...messageParts: TerminalWriteParameters): void {
    const { parts } = this._normalizeWriteParameters(messageParts);
    this._writeSegmentsToProviders(parts, TerminalProviderSeverity.debug, true);
  }

  private _writeSegmentsToProviders(
    segments: string[],
    severity: TerminalProviderSeverity,
    followedByEol: boolean
  ): void {
    const lines: string[] = [segments.join('')];
    if (followedByEol) {
      lines.push('');
    }

    let linesWithoutColor: string[] | undefined;

    const concatenatedLinesWithColorByNewlineChar: Map<string, string> = new Map();
    const concatenatedLinesWithoutColorByNewlineChar: Map<string, string> = new Map();
    for (const provider of this._providers) {
      let textToWrite: string | undefined;
      const eol: string = provider.eolCharacter;
      if (provider.supportsColor) {
        textToWrite = concatenatedLinesWithColorByNewlineChar.get(eol);
        if (!textToWrite) {
          textToWrite = lines.join(eol);
          concatenatedLinesWithColorByNewlineChar.set(eol, textToWrite);
        }
      } else {
        textToWrite = concatenatedLinesWithoutColorByNewlineChar.get(eol);
        if (!textToWrite) {
          if (!linesWithoutColor) {
            linesWithoutColor = [];
            for (const line of lines) {
              linesWithoutColor.push(AnsiEscape.removeCodes(line));
            }
          }

          textToWrite = linesWithoutColor.join(eol);
          concatenatedLinesWithoutColorByNewlineChar.set(eol, textToWrite);
        }
      }

      provider.write(textToWrite, severity);
    }
  }

  private _normalizeWriteParameters(parameters: TerminalWriteParameters): {
    parts: string[];
    options: ITerminalWriteOptions;
  } {
    if (parameters.length === 0) {
      return { parts: [], options: {} };
    } else {
      const lastParameter: string | ITerminalWriteOptions = parameters[parameters.length - 1];
      if (typeof lastParameter === 'string') {
        return { parts: parameters as string[], options: {} };
      } else {
        return { parts: parameters.slice(0, -1) as string[], options: lastParameter };
      }
    }
  }
}
