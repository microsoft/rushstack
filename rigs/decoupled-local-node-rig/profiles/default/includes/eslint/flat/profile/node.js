// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

// This profile enables lint rules intended for a general Node.js project, typically a web service.
// It enables security rules that assume the service could receive malicious inputs from an
// untrusted user.  If that is not the case, consider using the "node-trusted-tool" profile instead.

const nodeProfile = require('@rushstack/eslint-config/flat/profile/node');

const { localCommonConfig } = require('./_common');

module.exports = [...nodeProfile, ...localCommonConfig];
