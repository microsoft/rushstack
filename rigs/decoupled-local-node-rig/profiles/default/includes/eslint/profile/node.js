// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

// This profile enables lint rules intended for a general Node.js project, typically a web service.
// It enables security rules that assume the service could receive malicious inputs from an
// untrusted user.  If that is not the case, consider using the "node-trusted-tool" profile instead.

const { buildRules } = require('./_common');

const rules = buildRules('node');
module.exports = rules;
