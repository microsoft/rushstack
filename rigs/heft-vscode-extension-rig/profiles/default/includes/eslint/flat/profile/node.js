// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

const nodeProfile = require('@rushstack/heft-node-rig/profiles/default/includes/eslint/flat/profile/node');

module.exports = [
  ...nodeProfile,
  {
    files: ['**/*.ts', '**/*.tsx'],
    rules: {
      // Rationale: Use of `void` to explicitly indicate that a floating promise is expected
      // and allowed.
      'no-void': ['error', { allowAsStatement: true }]
    }
  }
];
