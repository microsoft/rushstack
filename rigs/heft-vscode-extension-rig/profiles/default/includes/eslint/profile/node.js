// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

module.exports = {
  extends: ['@rushstack/heft-node-rig/profiles/default/includes/eslint/profile/node'],

  rules: {
    // Rationale: Use of `void` to explicitly indicate that a floating promise is expected
    // and allowed.
    'no-void': ['error', { allowAsStatement: true }]
  }
};
