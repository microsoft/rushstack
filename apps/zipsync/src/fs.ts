// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { default as fs, type OpenMode } from 'node:fs';

interface IInternalDisposableFileHandle extends Disposable {
  fd: number;
}

export interface IDisposableFileHandle extends IInternalDisposableFileHandle {
  readonly fd: number;
}

export const DISPOSE_SYMBOL: typeof Symbol.dispose = Symbol.dispose ?? Symbol.for('Symbol.dispose');

export function getDisposableFileHandle(path: string, openMode: OpenMode): IDisposableFileHandle {
  const result: IInternalDisposableFileHandle = {
    fd: fs.openSync(path, openMode),
    [DISPOSE_SYMBOL]: () => {
      if (!isNaN(result.fd)) {
        fs.closeSync(result.fd);
        result.fd = NaN;
      }
    }
  };

  return result;
}

export function rmdirSync(dirPath: string): boolean {
  try {
    fs.rmdirSync(dirPath);
    return true;
  } catch (e) {
    if ((e as NodeJS.ErrnoException).code === 'ENOENT' || (e as NodeJS.ErrnoException).code === 'ENOTDIR') {
      // Not found, ignore
    } else {
      throw e;
    }
  }
  return false;
}

export function unlinkSync(filePath: string): boolean {
  try {
    fs.unlinkSync(filePath);
    return true;
  } catch (e) {
    if (e && (e as NodeJS.ErrnoException).code === 'ENOENT') {
      // Not found, ignore
    } else {
      throw e;
    }
  }
  return false;
}
