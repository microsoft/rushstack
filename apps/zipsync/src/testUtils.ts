// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { tmpdir } from 'node:os';
import * as path from 'node:path';
import * as fs from 'node:fs';
import * as crypto from 'node:crypto';

export function getTempDir(): string {
  const randomId: string = crypto.randomUUID();
  const tempDir: string = path.join(tmpdir(), `zipsync-test-${randomId}`);
  fs.mkdirSync(tempDir);
  return tempDir;
}

export function getDemoDataDirectoryDisposable(numFiles: number): {
  targetDirectories: string[];
  baseDir: string;
  [Symbol.dispose](): void;
} {
  const baseDir: string = getTempDir();

  const targetDirectories: string[] = ['demo-data-1', 'demo-data-2', 'demo-data-3', 'nested/demo/dir/4'].map(
    (folderName) => {
      const dataDir: string = path.join(baseDir, folderName);
      fs.mkdirSync(dataDir, { recursive: true });
      const subdir: string = path.join(dataDir, 'subdir');
      fs.mkdirSync(subdir);
      for (let i: number = 0; i < numFiles; ++i) {
        const filePath: string = path.join(subdir, `file-${i}.txt`);
        fs.writeFileSync(filePath, `This is file ${i} in ${folderName}/subdir\n`, { encoding: 'utf-8' });
      }
      return folderName;
    }
  );

  return {
    targetDirectories,
    baseDir,
    [Symbol.dispose]() {
      fs.rmSync(baseDir, { recursive: true, force: true });
    }
  };
}
