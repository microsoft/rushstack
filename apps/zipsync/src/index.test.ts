// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import * as path from 'node:path';
import * as fs from 'node:fs';

import { NoOpTerminalProvider } from '@rushstack/terminal/lib/NoOpTerminalProvider';
import { Terminal } from '@rushstack/terminal/lib/Terminal';

import { pack, unpack } from './index';
import { getDemoDataDirectoryDisposable } from './testUtils';

describe('zipSync tests', () => {
  it('basic pack test', () => {
    using demoDataDisposable = getDemoDataDirectoryDisposable(5);
    const { targetDirectories, baseDir } = demoDataDisposable;

    const terminal = new Terminal(new NoOpTerminalProvider());

    const archivePath: string = path.join(baseDir, 'archive.zip');
    const packResult = pack({
      terminal: terminal,
      compression: 'deflate',
      baseDir,
      targetDirectories,
      archivePath
    });

    expect(packResult).toMatchSnapshot();

    using unpackDemoDataDisposable = getDemoDataDirectoryDisposable(2);
    const { baseDir: unpackBaseDir } = unpackDemoDataDisposable;

    const unpackResult = unpack({
      terminal: terminal,
      archivePath,
      baseDir: unpackBaseDir,
      targetDirectories
    });

    expect(unpackResult).toMatchSnapshot();

    // Verify files were extracted
    for (const targetDirectory of targetDirectories) {
      const sourceDir: string = path.join(baseDir, targetDirectory);
      for (let i: number = 0; i < 5; ++i) {
        const sourceFile: string = path.join(sourceDir, 'subdir', `file-${i}.txt`);
        const destFile: string = path.join(unpackBaseDir, targetDirectory, 'subdir', `file-${i}.txt`);
        expect(fs.readFileSync(destFile, { encoding: 'utf-8' })).toEqual(
          fs.readFileSync(sourceFile, { encoding: 'utf-8' })
        );
      }
    }
  });
});
