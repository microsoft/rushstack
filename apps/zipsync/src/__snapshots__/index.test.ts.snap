// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zipSync tests basic pack test 1`] = `
Object {
  "filesPacked": 21,
  "metadata": Object {
    "files": Object {
      "demo-data-1/subdir/file-0.txt": Object {
        "sha1Hash": "a7f3d7debc1ad5f045f672be8a8e67d32c909fee",
        "size": 37,
      },
      "demo-data-1/subdir/file-1.txt": Object {
        "sha1Hash": "748119e4bafabea3b45d3f790c9a9b676ebab405",
        "size": 37,
      },
      "demo-data-1/subdir/file-2.txt": Object {
        "sha1Hash": "ace390b17191467c0a0480c264b6a2af0dff0d07",
        "size": 37,
      },
      "demo-data-1/subdir/file-3.txt": Object {
        "sha1Hash": "c7e2d154422af74d8b9242565e4a5988d5361b8c",
        "size": 37,
      },
      "demo-data-1/subdir/file-4.txt": Object {
        "sha1Hash": "f81f87be14a0c7d5e75ae0dd730526dc333fe1bd",
        "size": 37,
      },
      "demo-data-2/subdir/file-0.txt": Object {
        "sha1Hash": "423a13f056c5a2a3c5aaba7d8097af61f49819d5",
        "size": 37,
      },
      "demo-data-2/subdir/file-1.txt": Object {
        "sha1Hash": "0141eb229e83217fbf001e339ca58f0f0c820df1",
        "size": 37,
      },
      "demo-data-2/subdir/file-2.txt": Object {
        "sha1Hash": "31fb9ce64e21441701e44a7b9d2cc50446d55462",
        "size": 37,
      },
      "demo-data-2/subdir/file-3.txt": Object {
        "sha1Hash": "7e95ef682efc09c20871ae9e7abf8383b28b1fb1",
        "size": 37,
      },
      "demo-data-2/subdir/file-4.txt": Object {
        "sha1Hash": "744ee77039600639a4c2d50b0248ad39e2bc10d5",
        "size": 37,
      },
      "demo-data-3/subdir/file-0.txt": Object {
        "sha1Hash": "504cd07cb5b0d6dfc186247b34b2b960668a9977",
        "size": 37,
      },
      "demo-data-3/subdir/file-1.txt": Object {
        "sha1Hash": "b7b12f3f33565792d8563c23edaccd09b0f324ab",
        "size": 37,
      },
      "demo-data-3/subdir/file-2.txt": Object {
        "sha1Hash": "fab730bce23d9a87d3eb0b7bb30b992a47263505",
        "size": 37,
      },
      "demo-data-3/subdir/file-3.txt": Object {
        "sha1Hash": "36a9851256e27975b707ec335c9099ee4f70aac9",
        "size": 37,
      },
      "demo-data-3/subdir/file-4.txt": Object {
        "sha1Hash": "078d98ee46c668fefe397fa81b682ed24b70418b",
        "size": 37,
      },
      "nested/demo/dir/4/subdir/file-0.txt": Object {
        "sha1Hash": "70bcbc5ec0a9e2394edf1760dc2b57bff4f2486f",
        "size": 43,
      },
      "nested/demo/dir/4/subdir/file-1.txt": Object {
        "sha1Hash": "0977052a86aa8ccd3ee550218d529e34550a62d7",
        "size": 43,
      },
      "nested/demo/dir/4/subdir/file-2.txt": Object {
        "sha1Hash": "cd170868740762e98657d851724406b494581030",
        "size": 43,
      },
      "nested/demo/dir/4/subdir/file-3.txt": Object {
        "sha1Hash": "6de572afa4315a597a0a43e5636a228573993da5",
        "size": 43,
      },
      "nested/demo/dir/4/subdir/file-4.txt": Object {
        "sha1Hash": "c391c02c9bc1139bee8452c911bce484497169ba",
        "size": 43,
      },
    },
    "version": "1.0",
  },
}
`;

exports[`zipSync tests basic pack test 2`] = `
Object {
  "filesDeleted": 0,
  "filesExtracted": 12,
  "filesSkipped": 8,
  "foldersDeleted": 0,
  "metadata": Object {
    "files": Object {
      "demo-data-1/subdir/file-0.txt": Object {
        "sha1Hash": "a7f3d7debc1ad5f045f672be8a8e67d32c909fee",
        "size": 37,
      },
      "demo-data-1/subdir/file-1.txt": Object {
        "sha1Hash": "748119e4bafabea3b45d3f790c9a9b676ebab405",
        "size": 37,
      },
      "demo-data-1/subdir/file-2.txt": Object {
        "sha1Hash": "ace390b17191467c0a0480c264b6a2af0dff0d07",
        "size": 37,
      },
      "demo-data-1/subdir/file-3.txt": Object {
        "sha1Hash": "c7e2d154422af74d8b9242565e4a5988d5361b8c",
        "size": 37,
      },
      "demo-data-1/subdir/file-4.txt": Object {
        "sha1Hash": "f81f87be14a0c7d5e75ae0dd730526dc333fe1bd",
        "size": 37,
      },
      "demo-data-2/subdir/file-0.txt": Object {
        "sha1Hash": "423a13f056c5a2a3c5aaba7d8097af61f49819d5",
        "size": 37,
      },
      "demo-data-2/subdir/file-1.txt": Object {
        "sha1Hash": "0141eb229e83217fbf001e339ca58f0f0c820df1",
        "size": 37,
      },
      "demo-data-2/subdir/file-2.txt": Object {
        "sha1Hash": "31fb9ce64e21441701e44a7b9d2cc50446d55462",
        "size": 37,
      },
      "demo-data-2/subdir/file-3.txt": Object {
        "sha1Hash": "7e95ef682efc09c20871ae9e7abf8383b28b1fb1",
        "size": 37,
      },
      "demo-data-2/subdir/file-4.txt": Object {
        "sha1Hash": "744ee77039600639a4c2d50b0248ad39e2bc10d5",
        "size": 37,
      },
      "demo-data-3/subdir/file-0.txt": Object {
        "sha1Hash": "504cd07cb5b0d6dfc186247b34b2b960668a9977",
        "size": 37,
      },
      "demo-data-3/subdir/file-1.txt": Object {
        "sha1Hash": "b7b12f3f33565792d8563c23edaccd09b0f324ab",
        "size": 37,
      },
      "demo-data-3/subdir/file-2.txt": Object {
        "sha1Hash": "fab730bce23d9a87d3eb0b7bb30b992a47263505",
        "size": 37,
      },
      "demo-data-3/subdir/file-3.txt": Object {
        "sha1Hash": "36a9851256e27975b707ec335c9099ee4f70aac9",
        "size": 37,
      },
      "demo-data-3/subdir/file-4.txt": Object {
        "sha1Hash": "078d98ee46c668fefe397fa81b682ed24b70418b",
        "size": 37,
      },
      "nested/demo/dir/4/subdir/file-0.txt": Object {
        "sha1Hash": "70bcbc5ec0a9e2394edf1760dc2b57bff4f2486f",
        "size": 43,
      },
      "nested/demo/dir/4/subdir/file-1.txt": Object {
        "sha1Hash": "0977052a86aa8ccd3ee550218d529e34550a62d7",
        "size": 43,
      },
      "nested/demo/dir/4/subdir/file-2.txt": Object {
        "sha1Hash": "cd170868740762e98657d851724406b494581030",
        "size": 43,
      },
      "nested/demo/dir/4/subdir/file-3.txt": Object {
        "sha1Hash": "6de572afa4315a597a0a43e5636a228573993da5",
        "size": 43,
      },
      "nested/demo/dir/4/subdir/file-4.txt": Object {
        "sha1Hash": "c391c02c9bc1139bee8452c911bce484497169ba",
        "size": 43,
      },
    },
    "version": "1.0",
  },
  "otherEntriesDeleted": 0,
}
`;
